### ISSUE TITLE
Compatibility with woocommerce_get_price_html filter

### ISSUE THREAD

**Issue by vyskoczilova** [woocommerce/woocommerce-gutenberg-products-block#9374]
## Describe the bug

We want to use both PHP parts of WooCommerce and new blocks and found out that the `woocommerce_get_price_html` filter doesn't have any effect on blocks (for example, display all blocks). Is there a way to port easily this feature (and keep changes at just one place)? Thanks!

## Expected behavior

Existing "old" PHP filters are working both in blocks and PHP version of WooCommerce.

### WordPress (please complete the following information):

-   WordPress version: 6.2
-   WooCommerce version: 7.6.1
-   WooCommerce Blocks version: 10.1.0

**Comment by nielslange** [comment]
Thanks for reaching out, @vyskoczilova. To better understand your use case, would you mind explaining what you are hoping to accomplish with the `woocommerce_get_price_html` filter in the various blocks?

**Comment by vyskoczilova** [comment]
Hi! For example, we want to show prices both with VAT and without VAT; normally, we use this filter. However, it doesn't get through the blocks (and I guess there will be other usages as well).

**Comment by nielslange** [comment]
Thanks for providing additional context. I understand that you want to use the `woocommerce_get_price_html` filter to display prices both with and without VAT. Can you tell me which blocks should be affected by that? Do you have a working example, using the classic WooCommerce shortcodes, to see what outcome you have in mind?

**Comment by vyskoczilova** [comment]
All blocks that display products and their prices (all products, category listings, etc., single products price..) - ideally very same and consistent with the effect with `woocommerce_get_price_html`. 

Here is a temp site: https://sniffstone.s1-tastewp.com/shop/ default view (together with categories etc.) works. But here, if I place some of the blocks: https://sniffstone.s1-tastewp.com/wc-shortcodes/ it doesn't.

Hope it helps

**Comment by mikejolley** [comment]
It's likely a new method of changing the pricing will be needed so it can be done on the client side. This one however includes more than just cart and checkout since it needs to also apply to product blocks.

We'll be reviewing extensibility points in the coming weeks so I'll flag this for review.

**Comment by ralucaStan** [comment]
@pmcpinto is this something that we would like to have supported as part of the extensibility project? pinging  @chrisdesrochers  for awareness

**Comment by pmcpinto** [comment]
> @pmcpinto is this something that we would like to have supported as part of the extensibility project? pinging @chrisdesrochers for awareness

I think it should be included in the analysis to identify effort and priority. When it comes to implementation, we should talk to the Store Editing teams, as this goes beyond the checkout blocks.

**Comment by gigitux** [comment]
On the last months we worked on the Products block (and Product Collection block) to increase the compatibility with the actions/filter of WooCommerce Core. `woocommerce_get_price_html` is supported. #8854 

I added this PHP to my website script:

```
add_filter( 'woocommerce_get_price_html', function() {
	return 2;
}, 10, 2 );
```

I added the Products block in a post and on the frontend I see this:

![image](https://github.com/woocommerce/woocommerce-blocks/assets/4463174/43f1b5e2-da78-42bc-bf6a-83d6fe0f2f54)


In the future we are going to the deprecated all the Products Grid except for the Products block (in the future it will called Product Collection). 

I will close this issue, but feel free to continue the discussion if something isn't clear.
